hudi-common/src/main/java/org/apache/hudi/common/util/FileIOUtils.java

```diff
@@ -91,4 +94,29 @@ public static void writeStringToFile(String str, String filePath) throws IOExcep
     out.flush();
     out.close();
   }
+
+  /**
+   * Closes a {@link Closeable}, with control over whether an {@code IOException} may be thrown.
+   * @param closeable the {@code Closeable} object to be closed, or null,
+   *      in which case this method does nothing.
+   * @param swallowIOException if true, don't propagate IO exceptions thrown by the {@code close} methods.
+   *
+   * @throws IOException if {@code swallowIOException} is false and {@code close} throws an {@code IOException}.
+   */
+  public static void close(@Nullable Closeable closeable, boolean swallowIOException)
+      throws IOException {
+    if (closeable == null) {
+      return;
+    }
+    try {
+      closeable.close();
+    } catch (IOException e) {
+      if (!swallowIOException) {
+        throw e;
+      }
+    }
+  }
+
+  /** Maximum loop count when creating temp directories. */
+  private static final int TEMP_DIR_ATTEMPTS = 10000;
```

Comment:
we don't need this anymore correct?

Reply:
I think we do - this is getting called in Metrics.java - but I guess it could be dispensed with if we modify Metrics.java - haven't looked at it closely enough - will take that up.